### ISSUE TITLE
version_control lecture: 

### ISSUE THREAD

**Issue by PooyaFa** [QuantEcon/lecture-source-jl#284]

**Comment by arnavs** [comment]
Responses: 

* Reading/Reverting- Could you elaborate? The image provided shows a revert commit (your "commit 2" ) in the history, stacked on top of the old commit (your "commit 1"). And it shows the content of the revert (i.e., undoing the change we had previously committed). The Markdown text also says "this will create the inverse commit."

* Merge conflicts- What was confusing? It shows the little warning triangle in the file with the merge conflict, and then the content of the merge conflict. 

* VS Code/Atom- Same as above. I just put a direct screen grab of what you'd get if you opened the file with the merge conflict in Atom. As for the Atom workflow, I think we'd put the `tools_editors` lecture before this, but that's a good point.

* Command Line- Did you work through the [linked tutorial](https://www.git-tower.com/learn/git/ebook/en/command-line/appendix/command-line-101)? This is an optional section (and says as much in the Markdown) that we put in because it comes up often enough. The [git flight rules repo](https://github.com/k88hudson/git-flight-rules/) repo we linked above I think is the most detailed reference I've seen, but happy to add a new one. 

tl;dr Happy to make changes, but it would help if you could elaborate on what's not missing in the image, or what part of the workflow you were confused by.

**Comment by PooyaFa** [comment]
- Reading/Reverting: If I did as the lecture states, then commit 2 after reverting will be gone, and it seems that I can't back to it. What if I want to revert from commit 2 to 1 and then make some changes and then commit 3 with another branch! Therefore, it is good if the lecture states what will happen to commit 2 after we reverting it to commit 1. is it the same as "discard changes"?

- Merge conflicts: I can't see the content of the merge conflict.

- VS Code/Atom: I can't see the content here either

- For a quick start, I think maybe it is a good idea for a beginner to learn what is a "branch", "upstream", what happens when we "push"... the links you referred are either too easy for a start or too complicated, in my opinion. That's why I think http://marklodato.github.io/visual-git-guide/index-en.html is maybe better. Another option, I think, might be to explain another codes in command-line like merge, rebase, revert, add, and commit (the ones I think more likely to be needed for the beginners)!

**Comment by arnavs** [comment]
Hey @PooyaFa, just to be clear-

1. If you did as the lecture states, both the original commit and the revert commit will still exist in the history. You could “go back to it” like any other commit, but we didn’t talk about that workflow here (just covered resetting against the server). You can actually go to the link and see an example of this. So I’m curious about why this didn’t come through in the lecture and images. Again, not to say that this isn’t useful feedback, but I’m just having trouble seeing why, exactly, the content isn’t connecting. Also, I don’t think we need to worry about branches here.

2. We can definitely make the images bigger, but the key isn’t for you to read the text exactly. Rather, it’s to show you the workflow. 

3. We will be reorganizing the quick start. But, I don’t think we need to touch branches until absolutely necessary, and I think the thing you posted with the history diagrams is way too complicated. Remember, this is workflow-based — there’s no point showing people what’s happening “in principle,” if it’s irrelevant to their work.

**Comment by arnavs** [comment]
Also, I’m hesitant to dive too deeply into the CLI because we are really targeting the Desktop GUI. The flight rules repo though covers every conceivable scenario.

**Comment by PooyaFa** [comment]
My point is not that I couldn't figure it out. We have two options: either explain what will happen if reader reverts, or let the reader figure it out! I thought, maybe it is a good idea to explain it. 

Again, we can let the reader see for herself what will happen if she opens a conflicted file and how it looks like. Or, we could zoom in a little bit, and then she see exactly what happens.

This problem is what I am facing now! I prefer to know what is branches instead of just push and pull, because I am new to this and already made many mistakes and accidentally did wrong stuff git. However, since I've started learning the branches and understand what's going on, I feel more comfortable with the codes in git. This is I think only my point of view, I don't know if this is a general thing.

**Comment by arnavs** [comment]
@PooyaFa Let's back up a bit. I think we're on the same page, but when I talk about "[something] coming through in the lecture and images," or about "the content...connecting," I'm talking about how this would play with a complete novice reader. So not about your getting it or not, but rather about how a hypothetical beginner would learn.

That said, 

1. Reverting- This is what we have. We also work with the revert commit later. If you go to the https://github.com/quanteconuser/example_repository, which is the accompaniment to this, you'll see that one of the 4 commits is the reversion. So I'm just curious about why it seems that the commit would "disappear."

<img width="1253" alt="screen shot 2018-11-11 at 12 26 43 pm" src="https://user-images.githubusercontent.com/5009477/48317910-8e0b2c80-e5ad-11e8-8559-d0adf7548812.png">

2. Images- No issues with zooming in. I'll play around with the settings and see if we can snip out irrelevant whitespace. 

3. > I  prefer to know what are branches instead of just push and pull

     I think your use case (collaborating here) is already more advanced than most people we're targeting. Nine out of ten times, for individual projects or those with a few collaborators, branches are irrelevant. I agree with what you're saying, that learning the theory makes it much less mysterious and more approachable. But in terms of the average user, the goal is to give them an ultraminimal starter.

**Comment by arnavs** [comment]
Also, the goal is to really stay away from the command line `git` interface. I think the main argument for putting it in is not that people should be using it, but that enough people ask that we should address it. Most of the stuff we present here and elsewhere (like in the unit testing/CI) is based on the desktop GUI. It's just not worth the overhead of teaching two parallel ways of doing the same thing, IMO.

**Comment by arnavs** [comment]
Closing this because I added another disclaimer about the reversions.
